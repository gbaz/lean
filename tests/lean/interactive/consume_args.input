VISIT consume_args.lean
SYNC 7
import logic data.nat.basic
open nat eq.ops algebra

theorem tst (a b c : nat) : a + b + c = a + c + b :=
calc a + b + c = a + (b + c) : by rewrite add.assoc
         ...   = a + (c + b) : by rewrite (add.comm b c)
         ...   = a + c + b   : by rewrite add.assoc
WAIT
INFO 7