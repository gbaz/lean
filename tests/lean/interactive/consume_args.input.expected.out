-- BEGINWAIT
-- ENDWAIT
-- BEGININFO
-- SYMBOL|7|15
=
-- ACK
-- TYPE|7|17
ℕ
-- ACK
-- IDENTIFIER|7|17
a
-- ACK
-- TYPE|7|19
ℕ → ℕ → ℕ
-- ACK
-- SYMBOL|7|19
+
-- ACK
-- TYPE|7|21
ℕ
-- ACK
-- IDENTIFIER|7|21
c
-- ACK
-- TYPE|7|23
ℕ → ℕ → ℕ
-- ACK
-- SYMBOL|7|23
+
-- ACK
-- TYPE|7|25
ℕ
-- ACK
-- IDENTIFIER|7|25
b
-- ACK
-- SYMBOL|7|31
by
-- ACK
-- SYMBOL|7|34
rewrite
-- ACK
-- TYPE|7|42
∀ (a_1 b_1 c_1 : ?A), (:a_1 + b_1 + c_1:) = (:a_1 + (b_1 + c_1):)
-- ACK
-- IDENTIFIER|7|42
add.assoc
-- ACK
-- ENDINFO
